File: src/models/video.py
"""Data models for the video streaming platform."""
from pydantic import BaseModel
from typing import Optional, List, Dict, Any
from datetime import datetime
from enum import Enum


class VideoStatus(str, Enum):
    """Video processing status."""
    UPLOADED = "uploaded"
    INDEXING = "indexing"
    INDEXED = "indexed"
    FAILED = "failed"


class Video(BaseModel):
    """Video model."""
    id: str
    name: str
    blob_url: str
    status: VideoStatus
    uploaded_at: datetime
    indexed_at: Optional[datetime] = None
    duration: Optional[float] = None
    size_bytes: Optional[int] = None
    content_type: Optional[str] = None


class VideoInsights(BaseModel):
    """Video insights from Video Indexer."""
    video_id: str
    transcript: Optional[str] = None
    keywords: List[str] = []
    topics: List[str] = []
    faces: List[Dict[str, Any]] = []
    labels: List[str] = []
    sentiments: List[Dict[str, Any]] = []
    brands: List[str] = []
    language: Optional[str] = None


class VideoUploadResponse(BaseModel):
    """Response for video upload."""
    video_id: str
    blob_url: str
    message: str


class VideoListResponse(BaseModel):
    """Response for listing videos."""
    videos: List[Video]
    total: int


class AnalyticsData(BaseModel):
    """Analytics data model."""
    total_videos: int
    total_duration: float
    indexed_videos: int
    failed_videos: int
    top_keywords: List[Dict[str, Any]]
    top_topics: List[Dict[str, Any]]

File: web/lib/schemas/video.ts
/**
 * Video domain types mirroring the FastAPI backend Pydantic models.
 */

export type VideoStatus = "uploaded" | "indexing" | "indexed" | "failed";

export interface Video {
  id: string;
  name: string;
  blob_url: string;
  status: VideoStatus;
  uploaded_at: string;
  indexed_at?: string;
  duration?: number;
  size_bytes?: number;
  content_type?: string;
}

export interface VideoInsights {
  video_id: string;
  transcript?: string;
  keywords: string[];
  topics: string[];
  faces: Record<string, unknown>[];
  labels: string[];
  sentiments: Record<string, unknown>[];
  brands: string[];
  language?: string;
}

export interface VideoUploadResponse {
  video_id: string;
  blob_url: string;
  message: string;
}

export interface VideoListResponse {
  videos: Video[];
  total: number;
}

export interface VideoIndexResponse {
  message: string;
  video_id: string;
  status: VideoStatus;
}

export interface VideoDeleteResponse {
  message: string;
}
